pocs/filterwheel/filterwheel.py

```diff
@@ -0,0 +1,224 @@
+            a Quantity with time units. If a numeric type without units is given seconds will be
+            assumed. Default is None (no timeout).
+        serial_number (str, optional): serial number of the filter wheel, default 'XXXXXX'
+    """
+    def __init__(self,
+                 name='Generic Filter Wheel',
+                 model='simulator',
+                 camera=None,
+                 filter_names=None,
+                 timeout=None,
+                 serial_number='XXXXXX',
+                 *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._model = model
+        self._name = name
+        self._camera = camera
+        self._filter_names = [str(name) for name in listify(filter_names)]
+        if not self._filter_names:
+            # Empty list
+            msg = "Must provide list of filter names"
+            self.logger.error(msg)
+            raise ValueError(msg)
+        self._n_positions = len(filter_names)
+        if isinstance(timeout, u.Quantity):
+            self._timeout = timeout.to(u.second).value
+        else:
+            self._timeout = timeout
+        self._serial_number = serial_number
+        self._connected = False
+
+        self.logger.debug('Filter wheel created: {}'.format(self))
+
+##################################################################################################
+# Properties
+##################################################################################################
+
+    @property
+    def model(self):
+        """ Model of the filter wheel """
+        return self._model
+
+    @property
+    def name(self):
+        """ Name of the filter wheel """
+        return self._name
+
+    @property
+    def uid(self):
+        """ A serial number of the filter wheel """
+        return self._serial_number
+
+    @property
+    def is_connected(self):
+        """ Is the filterwheel available """
+        return self._connected
+
+    @property
+    def camera(self):
+        """
+        Reference to the Camera object that the FilterWheel is assigned to, if any. A filter wheel
+        should only ever be assigned to one or zero Cameras!
+        """
+        return self._camera
+
+    @camera.setter
+    def camera(self, camera):
+        if self._camera:
+            self.logger.warning("{} assigned to {}, skipping attempted assignment to {}!",
+                                self, self.camera, camera)
+        else:
+            self._camera = camera
+
+    @property
+    def filter_names(self):
+        """ List of the names of the filters installed in the filter wheel """
+        return self._filter_names
+
+    @property
+    def n_positions(self):
+        """ Number of positions in the filter wheel """
+        return self._n_positions
+
+    @property
+    def position(self):
+        """ Current integer position of the filter wheel """
+        raise NotImplementedError
+
+    @position.setter
+    def position(self, position):
```

Comment:
The setter on the position, while convenient, seems somewhat dangerous to me. I'm not suggesting we do away with it but seems like you could easily accidentally set this.

Reply:
It's really just there as a convenience during interactive use (e.g. testing in an IPython shell). The intent is that you almost always use the `.move_to()` method directly so that you can collect the events and move multiple filter wheels in parallel with each other. Perhaps the documentation should say that.